imports/client/components/FancyEditor.tsx

```diff
@@ -0,0 +1,639 @@
+  blockquote?: boolean;
+}
+
+interface MentionRendererProps extends ElementRendererProps<MentionElement> {
+  users: Map<string, Meteor.User>;
+}
+
+export const MentionSpan = styled.span<{
+  isSelf: boolean;
+}>`
+  padding: 2px 3px 3px;
+  height: 20px;
+  margin: 0 1px;
+  vertical-align: baseline;
+  display: inline-block;
+  border-radius: 4px;
+  color: #4649ef;
+  background-color: #ced0ed;
+  ${({ isSelf }) => isSelf && css`
+    background-color: #4649ef;
+    color: #ffffff;
+  `}
+  font-size: 0.8rem;
+`;
+
+const SelectedMentionSpan = styled(MentionSpan)`
+  box-shadow: 0 0 0 2px #b4d5ff;
+`;
+
+const EditableMentionRenderer = ({
+  attributes,
+  children,
+  element,
+  users,
+}: MentionRendererProps) => {
+  const selected = useSelected();
+  const focused = useFocused();
+  const user = users.get(element.userId);
+  const Elem = selected && focused ? SelectedMentionSpan : MentionSpan;
+
+  return (
+    <Elem {...attributes} contentEditable={false}>
+      {children}
+      @
+      {`${user?.displayName ?? element.userId}`}
+    </Elem>
+  );
+};
+
+// Composed, layered reassignment is how Slate plugins are designed to work.
+/* eslint-disable no-param-reassign */
+const withMentions = (editor: Editor) => {
+  const { isInline, isVoid, markableVoid } = editor;
+  editor.isInline = (element) => {
+    return element.type === 'mention' ? true : isInline(element);
+  };
+  editor.isVoid = (element) => {
+    return element.type === 'mention' ? true : isVoid(element);
+  };
+  editor.markableVoid = (element) => {
+    return element.type === 'mention' || markableVoid(element);
+  };
+  return editor;
+};
+
+// A normalization pass to merge all `message` elements into a single one.
+// Normal editing can't create more than one, but copy-pasting a break would
+// normally get transformed into a hard break, so we solve this by merging
+// sibling message nodes.
+const withSingleMessage = (editor: Editor) => {
+  const { normalizeNode } = editor;
+  editor.normalizeNode = (entry: NodeEntry) => {
+    const [node, _path] = entry;
+    if (Editor.isEditor(node)) {
+      if (node.children.length > 1) {
+        Transforms.mergeNodes(editor, { at: [1] });
+      }
+    }
+
+    normalizeNode(entry);
+  };
+  return editor;
+};
+/* eslint-enable no-param-reassign */
+
+const insertMention = (editor: Editor, userId: string) => {
+  const mention: MentionElement = {
+    type: 'mention',
+    userId,
+    children: [{ text: '' }],
+  };
+  Transforms.insertNodes(editor, mention);
+  Transforms.move(editor);
+};
+
+const MatchCandidate = ({
+  user, selected, onSelected,
+}: {
+  user: Meteor.User,
+  selected: boolean,
+  onSelected: (u: Meteor.User) => void,
+}) => {
+  const onClick = useCallback(() => {
+    onSelected(user);
+  }, [onSelected, user]);
+  return (
+    <div
+      key={user._id}
+      style={{
+        padding: '1px 3px',
+        borderRadius: '3px',
+        background: selected ? '#e0ecfc' : 'transparent',
+      }}
```

Comment:
Should this be a styled component?

Reply:
Yes.  Addressed in 398e9b23095f36dab44b10bd1a73983d5eed48ff when I made autocomplete show avatars